test/api/RegistrationTokenTest.ts

```diff
@@ -0,0 +1,175 @@
+        await testData.adminCentralService.checkEntityInList(
+          testData.adminCentralService.registrationApi,
+          testData.newRegistrationToken
+        );
+      });
+
+      // Update / revoke
+      it('Should be able to update a registration token', async () => {
+        testData.newRegistrationToken.description = faker.random.word();
+        testData.newRegistrationToken.expirationDate = faker.date.future();
+
+        await testData.adminCentralService.updateEntity(
+          testData.adminCentralService.registrationApi,
+          testData.newRegistrationToken,
+        );
+      });
+
+      // Verify update readById
+      it('Should find the updated registration token by id', async () => {
+        // Check if the updated entity can be retrieved with its id
+        const updatedRegistrationToken = await testData.adminCentralService.getEntityById(
+          testData.adminCentralService.registrationApi,
+          testData.newRegistrationToken,
+          false
+        );
+        expect(updatedRegistrationToken.data.description).to.equal(testData.newRegistrationToken.description);
+        expect(updatedRegistrationToken.data.expirationDate).to.equal(moment.utc(testData.newRegistrationToken.expirationDate).format('yyyy-MM-DD[T]HH:mm:ss.SSS[Z]'));
+      });
+
+      // Delete
+      it('Should be able to delete the created registration token', async () => {
+        // Delete the created entity
+        await testData.adminCentralService.deleteEntity(
+          testData.adminCentralService.registrationApi,
+          testData.newRegistrationToken
+        );
+      });
+
+      // Verify delete readById
+      it('Should not find the deleted asset with its id', async () => {
+        // Check the deleted entity cannot be retrieved with its id
+        await testData.adminCentralService.checkDeletedEntityById(
+          testData.adminCentralService.registrationApi,
+          testData.newRegistrationToken
+        );
+      });
+    });
+
+    describe('Where basic user', () => {
+      before(async () => {
+        testData.tenantContext = await ContextProvider.defaultInstance.getTenantContext(ContextDefinition.TENANT_CONTEXTS.TENANT_WITH_ALL_COMPONENTS);
+        testData.basicUserContext = testData.tenantContext.getUserContext(ContextDefinition.USER_CONTEXTS.BASIC_USER);
+        testData.basicCentralService = new CentralServerService(ContextDefinition.TENANT_CONTEXTS.TENANT_WITH_ALL_COMPONENTS, testData.basicUserContext);
+      });
+      // Create
+      it('Should not be able to create a new registration token', async () => {
+        // Check cannot create
+        testData.newRegistrationToken = Factory.registrationToken.build();
+        const response = await testData.basicCentralService.createEntity(
+          testData.basicCentralService.registrationApi,
+          testData.newRegistrationToken,
+          false
+        );
+        expect(response.status).to.be.eq(HTTPAuthError.ERROR);
```

Comment:
Ideally, you should return a HTTP 403 code.

Reply:
Done. They all return 403 now :-)